Source/lib/Result.cs

```diff
@@ -112,12 +149,18 @@ public Result(String text, byte[] rawBytes, ResultPoint[] resultPoints, BarcodeF
         /// <param name="format">The format.</param>
         /// <param name="timestamp">The timestamp.</param>
         public Result(String text, byte[] rawBytes, int numBits, ResultPoint[] resultPoints, BarcodeFormat format, long timestamp)
+            : this(text, null, rawBytes, numBits, resultPoints, format, timestamp)
+        {
+        }
+
+        public Result(String text, byte[] data, byte[] rawBytes, int numBits, ResultPoint[] resultPoints, BarcodeFormat format, long timestamp)
         {
             if (text == null && rawBytes == null)
             {
                 throw new ArgumentException("Text and bytes are null");
             }
             Text = text;
+            Data = data;

```

Comment:
this could be a good place to add the default behavior UTF8.GetBytes(text) if data == null